frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -271,42 +907,107 @@ def pack_to_chunk_groups_and_map(
 
-            y_offset (int, default=0):
+            y_offset (int, default = 0):
                 optional offset for x during chunk extraction
 
         Returns:
             List[Conversion]:
                 result of mapping chunks of x and y
         """
 
+        # this function computes the following in FHE
+        # -------------------------------------------
+        # result = []
+        # for chunk_index, (chunk_start, chunk_end) in enumerate(chunk_ranges):
+        #     x_chunk = (x + x_offset).bits[chunk_start:chunk_end]
+        #     y_chunk = (y + y_offset).bits[chunk_start:chunk_end]
+        #     result.append(mapper(chunk_index, x_chunk, y_chunk))
+        # return result
+        # -------------------------------------------
+        # - x_chunk :: { bit_width: (2 * chunk_size), shape: x.shape }
+        # - y_chunk :: { bit_width: (2 * chunk_size), shape: y.shape }
+        # - results :: { bit_width: resulting_scalar_type.bit_width, shape: resulting_shape }
+
+        assert resulting_scalar_type.is_scalar
+
         result = []
-        for chunk_index, offset in enumerate(range(0, bit_width, chunk_size)):
-            bits_to_process = min(chunk_size, bit_width - offset)
-            right_shift_by = bit_width - offset - bits_to_process
-            mask = (2**bits_to_process) - 1
+        for chunk_index, (chunk_start, chunk_end) in enumerate(chunk_ranges):
+            chunk_size = chunk_end - chunk_start
+
+            shift_to_clear_lsbs = chunk_start
+            mask_to_clear_msbs = (2**chunk_size) - 1
+
+            x_chunk_lut = [
+                (((x + x_offset) >> shift_to_clear_lsbs) & mask_to_clear_msbs) << chunk_size
+                for x in range(2**x.bit_width)
+            ]
+            y_chunk_lut = [
+                (((y + y_offset) >> shift_to_clear_lsbs) & mask_to_clear_msbs)
+                for y in range(2**y.bit_width)
+            ]
+
+            x_chunk_is_constant = all(x == x_chunk_lut[0] for x in x_chunk_lut)
+            y_chunk_is_constant = all(y == x_chunk_lut[0] for y in y_chunk_lut)
+
+            if x_chunk_is_constant and y_chunk_is_constant:  # pragma: no cover
+                result.append(
+                    self.constant(
+                        self.i(resulting_scalar_type.bit_width + 1),
+                        mapper(chunk_index, x_chunk_lut[0] >> chunk_size, y_chunk_lut[0]),
+                    )
+                )
+                continue
 
-            chunk_x = self.tlu(
-                resulting_type,
-                x,
-                [
-                    ((((x + x_offset) >> right_shift_by) & mask) << bits_to_process)
-                    for x in range(2**bit_width)
-                ],
-            )
-            chunk_y = self.tlu(
-                resulting_type,
-                y,
-                [((y + y_offset) >> right_shift_by) & mask for y in range(2**bit_width)],
-            )
+            if x_chunk_is_constant:
```

Comment:
I would put a elif to see the condition are mutually exclusives.

Reply:
Ruff doesn't like it:
```
RET507 Unnecessary `elif` after `continue` statement
```

Reply:
ruff is wrong half the time :)